src/main/java/com/sprint/mission/discodeit/controller/ChannelController.java

```diff
@@ -0,0 +1,96 @@
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/channels")
+public class ChannelController {
+
+    private final ChannelService channelService;
+
+    @PostMapping("/public")
+    public ResponseEntity<Object> createPublicChannel(
+        @RequestBody PublicChannelCreateRequest publicChannelCreateRequest) {
+        try {
+            return ResponseEntity.status(HttpStatus.CREATED)
+                .body(channelService.create(publicChannelCreateRequest));
+        } catch (IllegalArgumentException e) {
+            return ResponseEntity.badRequest().body(e.getMessage());
+        } catch (RuntimeException e) {
+            return ResponseEntity.internalServerError().body(e.getMessage());
+        }
+    }
+
+    @PostMapping("/private")
+    public ResponseEntity<Object> createPrivateChannel(
+        @RequestBody PrivateChannelCreateRequest privateChannelCreateRequest) {
+        try {
+            return ResponseEntity.status(HttpStatus.CREATED)
+                .body(channelService.create(privateChannelCreateRequest));
+        } catch (IllegalArgumentException e) {
+            return ResponseEntity.badRequest().body(e.getMessage());
+        } catch (RuntimeException e) {
+            return ResponseEntity.internalServerError().body(e.getMessage());
+        }
+    }
+
+    @PatchMapping("/{channelId}")
+    public ResponseEntity<Object> updatePublicChannel(@PathVariable UUID channelId,
+        @RequestBody PublicChannelUpdateRequest publicChannelUpdateRequest) {
+        try {
+            return ResponseEntity.ok(channelService.update(channelId, publicChannelUpdateRequest));
+        } catch (NoSuchElementException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                .body("Channel with id " + channelId + " not found: " + e.getMessage());
+        } catch (IllegalArgumentException e) {
+            return ResponseEntity.badRequest().body("Private channel cannot be updated");
+        } catch (RuntimeException e) {
+            return ResponseEntity.internalServerError().body(e.getMessage());
+        }
+    }
+
+    @DeleteMapping("/{channelId}")
+    public ResponseEntity<Object> removeChannel(@PathVariable UUID channelId) {
+        try {
+            channelService.delete(channelId);
+            return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+        } catch (NoSuchElementException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                .body("Channel with id " + channelId + " not found.");
+        } catch (IllegalArgumentException e) {
+            return ResponseEntity.badRequest().body(e.getMessage());
+        } catch (RuntimeException e) {
+            return ResponseEntity.internalServerError().body(e.getMessage());
+        }
+    }
+
+    @GetMapping("/")
```

Comment:
아래처럼 마지막에 슬래쉬("/") 빼야지만 프론트엔드에서 정상동작할 것 같아요.
```suggestion
    @GetMapping("")
```

지금은 마지막 슬래쉬때문에 정상동작하지 않네요.
<img width="515" height="300" alt="image" src="https://github.com/user-attachments/assets/97354b30-b0df-463b-80b2-34adaa15f00c" />